fastly/block_fastly_service_v1_logging_cloudfiles.go

```diff
@@ -0,0 +1,291 @@
+package fastly
+
+import (
+	"fmt"
+	"log"
+
+	gofastly "github.com/fastly/go-fastly/fastly"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+)
+
+type CloudfilesServiceAttributeHandler struct {
+	*DefaultServiceAttributeHandler
+}
+
+func NewServiceLoggingCloudfiles() ServiceAttributeDefinition {
+	return &CloudfilesServiceAttributeHandler{
+		&DefaultServiceAttributeHandler{
+			key: "logging_cloudfiles",
+		},
+	}
+}
+
+func (h *CloudfilesServiceAttributeHandler) Process(d *schema.ResourceData, latestVersion int, conn *gofastly.Client) error {
+	serviceID := d.Id()
+	ol, nl := d.GetChange(h.GetKey())
+
+	if ol == nil {
+		ol = new(schema.Set)
+	}
+	if nl == nil {
+		nl = new(schema.Set)
+	}
+
+	ols := ol.(*schema.Set)
+	nls := nl.(*schema.Set)
+
+	removeCloudfilesLogging := ols.Difference(nls).List()
+	addCloudfilesLogging := nls.Difference(ols).List()
+
+	// DELETE old Cloudfiles logging endpoints.
+	for _, oRaw := range removeCloudfilesLogging {
+		of := oRaw.(map[string]interface{})
+		opts := buildDeleteCloudfiles(of, serviceID, latestVersion)
+
+		log.Printf("[DEBUG] Fastly Cloudfiles logging endpoint removal opts: %#v", opts)
```

Comment:
Not sure if I should be using `Rackspace Cloud Files` or simply `Cloudfiles` as the docs and `manage.fastly.com` use both.

Reply:
Per their own product page, I'd go with "Cloud Files".
```suggestion
		log.Printf("[DEBUG] Fastly Cloud Files logging endpoint removal opts: %#v", opts)
```

Reply:
Similar comments elided for rest of review.